packages/webcert/src/store/certificate/certificateSelectors.ts

```diff
@@ -148,6 +148,29 @@ export const getCareUnitValidationErrors = () => (state: RootState): ValidationE
   return state.ui.uiCertificate.certificate.metadata.careUnitValidationErrors
 }
 
+const doesFieldsMatch = (payloadField: string, validationField: string) => {
+  return !validationField || validationField.includes(payloadField)
+}
+
+export const getVisibleValidationErrors = (questionId: string, field: string) => (state: RootState): ValidationError[] => {
```

Comment:
For performance reason memoize the clientValidation errors so that all the filtering logic doesn't have to be done if there is no change.

Here is a link: https://react-redux.js.org/api/hooks#using-memoizing-selectors

Reply:
Is it possible to do this when we need attributes from the component to do the filtering?

Reply:
@mww-sopra-steria

Reply:
@mww-sopra-steria Is it really possible to do use memoize when we need attributes from the component to do the filtering? If we don't have the questionId and field we can't filter out the validationMessages for the specific component

Reply:
I turn that question around a little. Have you found any documentation or tried and found that it won't work?

Reply:
@mww-sopra-steria https://stackoverflow.com/questions/40291084/use-reselect-selector-with-parameters

Reply:
It seems like there are ways to implement it, but I don't find it so intuitive or easy. I suggest you merge the PR and then I will analyze what the possible impact the current implementation has. If it is none to minimal, we can be satisfied with the current selector. If not it will be easier to experiment with the client by having all the changes integrated in the master branch.